### PR TITLE
Create nested mappings BigQuery indexer

### PR DESCRIPTION
**Pull request by bfcrampton**
For a BigQuery table with a schema definition like:
screen shot 2018-08-31 at 6 23 44 pm

This now creates the following nested field mappings in elastic search:
```
"mappings": {
  "type": {
    "properties": {
      "bryancrampton-testing": {
        "properties": {
          "bryan": {
            "properties": {
              "neseted_repeated_structss": {
                "properties": {
                  "repeated_structs": {
                    "type": "nested",
                    "properties": {
                      "parent": {
                        "type": "nested",
                        "properties": {
                          "child": {
                            "type": "nested"
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

As described [here](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/nested.html#_how_arrays_of_objects_are_flattened) object arrays are flattened which results in incorrect searching of inner fields of repeated structs. Adding these nested mappings fixes this issue.

### CODE REVIEW

File: bigquery/indexer.py [hunk 1]
```diff
@@ -5,18 +5,21 @@
...
 
-# Log to stderr.
 logging.basicConfig(
     level=logging.INFO,
     format='%(asctime)s %(filename)10s:%(lineno)s %(levelname)s %(message)s',
     datefmt='%Y%m%d%H:%M:%S')
 logger = logging.getLogger('indexer.bigquery')
 
 ES_TIMEOUT_SEC = 20
+SELECT_FORMAT = 'SELECT * FROM `%s`'
```
Comment (melissachang) [inline bigquery/indexer.py]: opt nit - For all 3 of these, especially SELECT_FORMAT, I find it more readable if they're inlined. Up to you.

File: bigquery/indexer.py [hunk 2]
```diff
@@ -48,6 +51,30 @@ def parse_args():
...
+        if field.mode == REPEATED_TYPE and field.field_type == STRUCT_TYPE:
+            name = '%s.%s' % (prefix, field.name) if prefix else field.name
+            nested[name] = {"type": "nested"}
+            inner_nested = get_nested_mappings(field.fields)
+            if inner_nested:
+                nested[name]['properties'] = inner_nested
+    return nested if nested else None
+
+
+def create_nested_mappings(es, index_name, table_name, billing_project_id):
```
Comment (melissachang) [inline bigquery/indexer.py]:
```
# Create nested mappings so queries will work correctly.
# See https://www.elastic.co/guide/en/elasticsearch/reference/6.4/nested.html#_how_arrays_of_objects_are_flattened
```

Comment (bfcrampton) [reply bigquery/indexer.py]: Done.

File: bigquery/indexer.py (hunk 1 above)
Comment (bfcrampton) [reply bigquery/indexer.py]: Done.

**Pull request merged**